### DISCUSSION TITLE
FileUpload no download

### DISCUSSION THREAD

**Discussion by eaziweb95** [filamentphp/filament#2954] [Help] [answered]
I use the FileUpload field. The file is well uploaded on the server under storage/app/public (see screen copy below) 
![file2](https://user-images.githubusercontent.com/58070424/176160175-3c6e8b8f-f3bd-42df-b990-8bb363a32aad.jpg)

but when i return on the form to download the file, i can't because the filed is in wait (see screen copy below). why ?
![file](https://user-images.githubusercontent.com/58070424/176160230-a8985580-7004-45a8-89e6-8f833409a7ad.jpg)

**Comment by studiowizjo** [comment, marked as answer]
Probably this; https://filamentphp.com/tricks/file-upload-previews-not-loading should help.

**Reply by eaziweb95** [reply]
I followed the instruction and change de APP_URL in .env. 
The field is no more in wait but i can't download the document to view it !
How can i do ?
![download](https://user-images.githubusercontent.com/58070424/176626877-a4c10c1e-84da-4ca4-9e1f-358bff7049b8.jpg)
?

**Reply by eaziweb95** [reply]
Thanks in advance for your reply

**Reply by chahal26** [reply]
Did you run `php artisan storage:link` ?

**Reply by studiowizjo** [reply]
I assume he did, because upload works now fine. Did you used `enableDownload()` method on your file definition?

```
FileUpload::make('attachment')
    ->enableDownload()
```

**Reply by eaziweb95** [reply]
no but when i add it, i have this message
Method Filament\Forms\Components\FileUpload::enableDownload does not exist.

**Reply by eaziweb95** [reply]
return Forms\Components\FileUpload::make('file_cv')->enableDownload()->required()
            ->label('CV')
            ->maxSize(2048)
            ->acceptedFileTypes(['application/pdf','application/msword','application/vnd.oasis.opendocument.text']);

**Reply by danharrin** [reply]
Because you aren't on the latest version

**Reply by eaziweb95** [reply]
![fileupload](https://user-images.githubusercontent.com/58070424/177165942-f3f11c9d-5852-40ea-a1df-f5085fff3cc5.jpg)

**Reply by eaziweb95** [reply]
"filament/filament": "^2.0",

**Reply by eaziweb95** [reply]
my composer.json files
    "require": {
        "php": "^8.0.2",
        "cossou/jasperphp": "^2.8",
        "doctrine/dbal": "^3.3",
        "filament/filament": "^2.0",
        "filament/forms": "^2.0",
        "guzzlehttp/guzzle": "^7.2",
        "illuminate/support": "^9.17",
        "laravel/framework": "^9.11",
        "laravel/sanctum": "^2.14.1",
        "laravel/tinker": "^2.7",
        "spatie/laravel-backup": "^8.1"
    },

**Reply by eaziweb95** [reply]
what is the issue ?

**Reply by eaziweb95** [reply]
i try to upgrade to filament 2.13, the enableDownload is well integrated and the download icon is displayed
![download](https://user-images.githubusercontent.com/58070424/177177101-92a4a9d8-17ef-4a3e-b1b6-88348d74e9c0.jpg)
 but when i click on it, no download, so i click on the right button on my mouse to download it but the file is not recognized
![download2](https://user-images.githubusercontent.com/58070424/177177549-bcfb84fe-b00d-469a-bbac-14dd75656b40.jpg)

**Reply by eaziweb95** [reply]
and i have another issue on the display. The menu navigation on the left is superimposed with the width of the page ... it is very embarrassing
![templatejpg](https://user-images.githubusercontent.com/58070424/177178268-4da26bb1-e660-4c33-b047-03227949abec.jpg)

**Reply by danharrin** [reply]
Do you have views in /resources/views/vendor/filament? Have you run `php artisan filament:upgrade`?

Do you have console errors for the file upload fields?

**Reply by studiowizjo** [reply]
Probably `npm run watch` should help, so js and css files will reflect changes after update.

**Reply by eaziweb95** [reply]
C:\xampp\htdocs\eva>npm run watch

> watch
> mix watch

'mix' n’est pas reconnu en tant que commande interne
ou externe, un programme exécutable ou un fichier de commandes.

Here is my packages.json

{
    "private": true,
    "scripts": {
        "dev": "npm run development",
        "development": "mix",
        "watch": "mix watch",
        "watch-poll": "mix watch -- --watch-options-poll=1000",
        "hot": "mix watch --hot",
        "prod": "npm run production",
        "production": "mix --production"
    },
    "devDependencies": {
        "axios": "^0.25",
        "laravel-mix": "^6.0.6",
        "lodash": "^4.17.19",
        "postcss": "^8.1.14"
    }
}

**Reply by studiowizjo** [reply]
For the future, please translate things to English. I suspect you haven't installed dependencies, so at the beginning run `npm run install` and then `npm run dev` or `npm run prod`.

**Reply by eaziweb95** [reply]
Another error
C:\xampp\htdocs\eva>npm run install
npm ERR! Missing script: "install"
npm ERR!
npm ERR! Did you mean this?
npm ERR!     npm uninstall # Remove a package
npm ERR!
npm ERR! To see a list of scripts, run:
npm ERR!   npm run

**Reply by studiowizjo** [reply]
Sorry, my bad - should be: `npm install`
